### DISCUSSION TITLE
eliminate the source.

### DISCUSSION THREAD

**Discussion by truonghoailinh47** [NanoComp/meep#2358] [General]
Dear Meep developer, I am trying to simulate electric field of nanoparticles. But in the final result, I always got the source line. Can you guys help me how to remove it? Below is my code and result. Thank you so much. 

import meep as mp
import numpy as np
import matplotlib.pyplot as plt
from matplotlib import cm
#radius
r = 25e-3
sizex  = 480e-3
sizey = 480e-3
sizez= 480e-3
size_3d=(sizex, sizey, sizez)
courant = 0.5
#pixels/um
resolution = 500
pixel_size = 1/resolution
plane_th = 1*r
#original thickness = 0.2*sizex
pml_th= 0.4
#pml_th= 2.5e-3
fullx = sizex + 2*pml_th
fully = sizey + 2*pml_th
fullz = sizez + 2*pml_th
cell = mp. Vector3(fullx, fully,0)
pml_layers = [mp.PML(thickness= pml_th, direction =mp.X), mp.PML(thickness= pml_th, direction =mp.Y),mp.PML(thickness=pml_th, direction =mp.Z)]

nonpml_vol = mp.Volume(mp.Vector3(),size = mp.Vector3(sizex, sizey, sizez))
freq = 0.473

comp= mp.Ex
sources = [mp.Source(mp.ContinuousSource(frequency=freq, is_integrated= True),
                     size= mp.Vector3(fullx, 0,0),
                     component= comp,
                    center = mp.Vector3(0,0))]

geometry = [mp.Sphere (material = mp. Medium(epsilon = 11.753, D_conductivity = 2*np.pi * 0.473*1.2696/11.753),
                       center = mp.Vector3(0,0,0),
                       radius=r),
            ]
sim = mp.Simulation (cell_size = cell,
                     boundary_layers = pml_layers,
                     geometry = geometry,
                     sources = sources,
                     resolution= resolution,
                     force_complex_fields= True,
                     split_chunks_evenly = False,)

plt.show()
![image](https://user-images.githubusercontent.com/112536838/209889803-a96b19a8-b982-4245-a88c-a7a16d125469.png)

**Comment by truonghoailinh47** [comment]
I am already done. Thank you.

**Comment by stevengj** [comment]
Well, you put a CW current source along the x axis, so your simulation will include this — Meep is doing what you tell it to do.

If you want to find a resonant mode, you should use a pulsed source (which turns off) and look at the fields some time later; see the tutorials.

**Reply by truonghoailinh47** [reply]
Thanks for your instruction. I will try.

**Reply by truonghoailinh47** [reply]
Dear Professor, 
I use Gaussian source for my simulation,I do not know how to choose exactly fwidth value?  In tutorials, its value show that No default value. You can instead specify fwidth=x .  I read some examples on tutorials, some examples multiply 0,2fcen, others examples multiply 0.1fcen. Because it will effect the final result, especially electric field intensity bar, that is a reason I have a little bit confuse.  Can you help me explain it? Thank you so much.

**Reply by stevengj** [reply]
The overall scaling is irrelevant because you are looking at the linear Maxwell equations. 

Usually, for resonant modes, you do two simulations — one broadband with harminv, to find all of the resonances in a bandwidth of interest, followed by one narrowband to visualize the fields.   In the first simulation, the `fwidth` is determined by the bandwidth of interest.  In the second simulation, you want the `fwidth` to be much smaller than the separation in frequency between nearby resonances, to be sure that you only excite one — but if you make it too narrow then the simulation will take a long time (proportional to 1/fwidth).

See for example this tutorial: https://meep.readthedocs.io/en/latest/Scheme_Tutorials/Resonant_Modes_and_Transmission_in_a_Waveguide_Cavity/

**Reply by truonghoailinh47** [reply]
> The overall scaling is irrelevant because you are looking at the linear Maxwell equations.
> 
> Usually, for resonant modes, you do two simulations — one broadband with harminv, to find all of the resonances in a bandwidth of interest, followed by one narrowband to visualize the fields. In the first simulation, the `fwidth` is determined by the bandwidth of interest. In the second simulation, you want the `fwidth` to be much smaller than the separation in frequency between nearby resonances, to be sure that you only excite one — but if you make it too narrow then the simulation will take a long time (proportional to 1/fwidth).
> 
> See for example this tutorial: https://meep.readthedocs.io/en/latest/Scheme_Tutorials/Resonant_Modes_and_Transmission_in_a_Waveguide_Cavity/

Thanks Professor.  Because I want to compare my simulation result (electric field intensity of nanoparticle) in Meep with Lumerical, that is why I mentioned about that scale bar.  I tried some simulation, same parameters ( like structure ,wavelength , position .....) with papers they used Lumerical. I got the nice result, but the scale bar is always different, sometimes lower , sometimes higher. Do you think that the result in Meep can match with result in Lumerical?

**Reply by smartalecH** [reply]
> but the scale bar is always different

As @stevengj mentioned, the overall scaling is irrelevant. The results of the "scale bar" could simply be due to a different source amplitude, for example (which is completely arbitrary). 

With linear simulations, we almost always look at _normalized_ responses for this very reason. If you're comparing your results to Lumerical or to a paper, see how they normalized their results, and make sure you do the same with meep.
